rxandroidble/src/main/java/com/polidea/rxandroidble/internal/operations/RxBleRadioOperationScan.java

```diff
@@ -34,20 +35,31 @@ public RxBleRadioOperationScan(UUID[] filterServiceUUIDs, RxBleAdapterWrapper rx
     @Override
     public void run() {
 
-        try {
-            boolean startLeScanStatus = rxBleAdapterWrapper.startLeScan(leScanCallback);
+        synchronized (isScanStarted) {
+
+            try {
 
-            if (!startLeScanStatus) {
-                onError(new BleScanException(BleScanException.BLUETOOTH_CANNOT_START));
+                isScanStarted.set(rxBleAdapterWrapper.startLeScan(leScanCallback));
+
+                if (!isScanStarted.get()) {
```

Comment:
I wanted to reuse a `status` object. But introducing additional one is an option too.